src/include/compat.h

```diff
@@ -179,7 +179,28 @@ struct cpu_set_t;
 #define MSG_DONTWAIT MSG_NONBLOCK
 #endif
 
-#if defined(HAVE_PTHREAD_SETNAME_NP)
+/* compiler warning free success noop */
+#define ceph_pthread_setname_noop(thread, name) ({ \
```

Comment:
I'd rename to something like `pthread_setname_noop_helper`, `pthread_getname_noop_helper` and `pthread_kill_unsupported_helper` so that these don't look like they are intended for direct use.